### ISSUE TITLE
More segfaults inside error handler

### ISSUE THREAD

**Issue by kirbyfan64** [kevinlawler/kona#326]
```
ryan@DevPC-LX:~/golf/hist$ k
K Console - Enter \ for help

  {[i]{[i;j]`0::[(~j)|(j=i)|j=n-1;"N";" "]}'!n;`0:"\n"}[i;]'!n
()
valence error
>  {[i]{[i;j]`0::[(~j)|(j=i)|j=n-1;"N";" "]}'!n;`0:"\n"}[i;]'!n
Segmentation fault (core dumped)
ryan@DevPC-LX:~/golf/hist$ 
```

**Comment by tavmem** [comment]
Interesting:
In both k2.8 and k3.2
```
 {[i]{[i;j]`0::[(~j)|(j=i)|j=n-1;"N";" "]}'!n;`0:"\n"}[i;]'!n
value error
i
^
parse error
```
Most importantly, k2.8 and k3.2 do **NOT** enter the error handler in this case.

**Comment by tavmem** [comment]
The fix for #324 also fixes this issue.
